src/elexmodel/handlers/data/Estimandizer.py

```diff
@@ -0,0 +1,113 @@
+        self.election_type = election_type
+        self.estimand_fns = estimand_fns
+        self.transformations = []
+        self.transformation_map = {
+            "party_vote_share": [self.calculate_party_vote_share],
+            "candidate": [self.candidate],
+        }
+
+    def pre_check_estimands(self):
+        """
+        Ensure estimand isn't one of the pre-specified values that are already included
+        """
+        standard = ["dem_votes", "gop_votes", "total_votes"]
+        if not self.check_input_columns(standard):
+            self.create_estimand(None, self.standard)
+
+    def check_input_columns(self, columns):
+        """
+        Check that input columns contain all neccessary values for a calculation
+        """
+        missing_columns = []
+        if self.election_type == "G":
+            missing_columns = [col for col in columns if col not in self.data_handler.data.columns]
+        elif self.election_type == "P":
+            missing_columns = [
+                col for col in columns if col not in self.data_handler.data[self.data_handler.election_id]
+            ]
+        return len(missing_columns) == 0
+
+    def verify_estimand(self, estimand):
+        """
+        Verify which estimands can be formed given a dataset and a list of estimands we would like to create
+        """
+        # Check if estimand is a supported value
+        if estimand not in self.transformation_map:
+            raise ValueError(f"Estimand '{estimand}' is not supported.")
+        self.transformations = self.transformation_map[estimand]
+
+        # Check if estimand function contains all the neccessary local variables to run
+        if not self.check_input_columns(
+            [col for transform in self.transformations for col in transform.__code__.co_varnames[1:]]
+        ):
+            return []
+
+        return self.transformations
+
+    def create_estimand(self, estimand=None, given_function=None):
+        """
+        Create an estimand. You must give either a estimand name or a pre-written function.
+        """
+        if estimand is None and given_function is not None:
+            given_function()
+        elif given_function is None and estimand is not None:
+            if estimand in self.transformation_map:
+                if self.transformation_map[estimand][0] in self.transformations:
+                    transformation_func = self.transformations[
+                        self.transformations.index(self.transformation_map[estimand][0])
+                    ]
+                    transformation_func()
+
+    def generate_estimands(self):
+        """
+        Main function to generate estimands
+        """
+        if self.election_type == "G":
+            self.pre_check_estimands()
+
+        for key, value in self.estimand_fns.items():
+            if value is None:  # Option 1: Pass in a list of estimands we want to build from a pre-set list
+                self.verify_estimand(key)
```

Comment:
how is `verify_estimand` related to `pre_check_estimands()`?

Reply:
`pre-check-estimands` just looks for and creates the standard values before any complex estimands are created, while `verify_estimand` is called to check if a given complex pre-written estimand is possible. these pre-written estimands usually depend on the existence of the standard estimands